atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/keyvalue/api/cache/LockWatchValueScopingCacheImpl.java

```diff
@@ -119,32 +118,31 @@ public synchronized void onSuccessfulCommit(long startTimestamp) {
         cache.finalise();
 
         Map<CellReference, CacheValue> cachedValues = cache.getValueDigest().loadedValues();
-        if (cachedValues.isEmpty()) {
-            return;
+        if (!cachedValues.isEmpty()) {
```

Comment:
drive-by fix: before, if there are no cached values, this wouldn't remove the state at the end! However, we did that as an extra step in a finally block in production, so this wasn't leaking memory anywhere outside of tests.